qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -487,22 +487,28 @@ def compute_minimum_eigenvalue(
         )
 
         start_time = time()
-        opt_params, opt_value, nfev = self.optimizer.optimize(
-            num_vars=len(initial_point),
-            objective_function=energy_evaluation,
-            gradient_function=gradient,
-            variable_bounds=bounds,
-            initial_point=initial_point,
-        )
+
+        # keep this until Optimizer.optimize is removed
+        try:
+            opt_result = self.optimizer.minimize(
+                fun=energy_evaluation, x0=initial_point, jac=gradient, bounds=bounds
+            )
+        except AttributeError:
+            # self.optimizer is an optimizer with the deprecated interface that uses
+            # ``optimize`` instead of ``minimize```
+            opt_result = self.optimizer.optimize(
+                len(initial_point), energy_evaluation, gradient, bounds, initial_point
+            )
```

Comment:
Do you think we should raise a warning here about using an optimizer with the old interface? This code will only run should a user have optimizer(s) that are not from the set provided in Terra. Now it would raise a warning if the optimizer itself calls super.optimizer, since a warning was added to base class, but if the derived 3rd party optimizer does not do that then a warning here would catch the usage and alert the user things were changing.

Reply:
Good point, added 👍🏻